src/main/java/com/griefcraft/modules/pluginsupport/Factions.java

```diff
@@ -99,4 +102,43 @@ public boolean canProtectLegacy(FPlayer fPlayer, Faction faction) throws Illegal
 
         return false;
     }
+
+    @Override
+    public void onAccessRequest(LWCAccessEvent event) {
+        Player player = event.getPlayer();
+        com.griefcraft.model.Protection protection = event.getProtection();
+
+        if (event.getAccess() != Permission.Access.NONE) {
+            return;
+        }
+
+        if (protection.getType() != com.griefcraft.model.Protection.Type.PRIVATE) {
+            return;
+        }
+
+        if (factions == null) {
+            return;
+        }
+
+        for (Permission permission : protection.getPermissions()) {
+            if (permission.getType() != Permission.Type.TOWN) {
```

Comment:
Probably shouldn't be town here
```suggestion
            if (permission.getType() != Permission.Type.FACTION) {
```